### PR TITLE
Feat: multifile prototype

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This will not pass `npm run test:curriculum`, but aside from that it should be useable.  I'll take it out of draft once the tests are passing.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
Please don't squash this - I'd like to keep the commit history until everything is done.

**Comment by ojeytonwilliams** [comment]
So,  https://github.com/freeCodeCamp/freeCodeCamp/pull/39223/commits/e012230141af94ef0de8b2c96398175da5acd152, fixes a bunch of challenges which had incorrect language tags and dummy solution code.  

The rationale was that we can then use the correct language tags to turn the solution into a 'file' of the appropriate type.  That makes single-file solutions a special case of multi-file solutions.

This PR is 'done', but I will clean things up tomorrow.

**Comment by RandellDawson** [comment]
The hint functionality appears to work great.  I merged in the WIP Cat Photo App branch and tested several steps and it always showed the text for the first test (in written test order) that failed.

**Review by raisedadead — DISMISSED** [review]
LGTM. @ojeytonwilliams please merge as you deem fit. There is a debug commit in there no sure what you want to do about it.

**Comment by ahmadabdolsaheb** [comment]
LGMT

**Review by ahmaxed — APPROVED** [review]

**Review by RandellDawson — APPROVED** [review]

File: curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-personal-portfolio-webpage.english.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ tests: []
 ## Solution
 <section id='solution'>
 
-```js
+```html
 // solution required
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-personal-portfolio-webpage.english.md:54]: @ojeytonwilliams Even though this is now HTML, is it a problem that the comment is a JavaScript comment?

File: tools/challenge-md-parser/solution-to-data.js [hunk 2]
```diff
@@ -2,17 +2,49 @@ const visit = require('unist-util-visit');
...
+          .map(indexByKey);
+
+        const solutions = isEmpty(solutionFiles)
+          ? solutionsAsFiles
+          : [solutionFiles];
+
         file.data = {
           ...file.data,
-          solutions
+          solutions: solutions
```
Comment (RandellDawson) [inline tools/challenge-md-parser/solution-to-data.js:47]: What was wrong with just `solutions` here?

File: curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-personal-portfolio-webpage.english.md (hunk 1 above)
Comment (ojeytonwilliams) [reply curriculum/challenges/english/01-responsive-web-design/responsive-web-design-projects/build-a-personal-portfolio-webpage.english.md:54]: Good catch, but no, it should not matter. It should just mean this challenge's tests should not run.

File: tools/challenge-md-parser/solution-to-data.js (hunk 2 above)
Comment (ojeytonwilliams) [reply tools/challenge-md-parser/solution-to-data.js:47]: I think that slipped through when I removed `solutionFiles` from the schema. It should just be `solution`

Comment (ojeytonwilliams) [reply tools/challenge-md-parser/solution-to-data.js]: I've fixed it.  Once you give the final go-ahead, I'll clean up the history a bit and we can merge.

Comment (ojeytonwilliams) [reply tools/challenge-md-parser/solution-to-data.js]: ...and I will remove the DEBUG challenge again.

**Pull request merged**